### ISSUE TITLE
Overflow size is calculated incorrectly in OracleDdlTable#createTable

### ISSUE THREAD

**Issue by hsaraogi** [palantir/atlasdb#1229]
From @clockfort :

okay it does set maxValueSize in TableMetadata, but it doesn't ever write it out anywhere

[10:00]  
it doesn't get passed into the toTableMetadata.

[10:04] ,
ooookay

[10:05]  
it gets recomputed from scratch in ColumnMetadataDescription after completely throwing out the one that the user manually sets

[10:05]  
 ```
    public int getMaxValueSize() {
        if (dynamicColumn != null) {
            return dynamicColumn.getValue().getMaxValueSize();
        }
        int max = 0;
        for (NamedColumnDescription col : namedColumns) {
            max = Math.max(max, col.getValue().getMaxValueSize());
        }
        return max;
    }
```
[10:10]  
This means we have a bug where any of our tables that have ValueTypes where it's not known at the valuetype level what the limiting size is (ValueType, Blob) will always have an overflow even if the table metadata says 'don't worry about it, the maximum size of that blob is less than 2000bytes'.

**Comment by clockfort** [comment]
I'm planning on fixing up at least the PG part of this for you, because it's going to want a PG upgrade task, and I imagine that you're already pretty disenchanted with the amount of extra work you've had to put into this.

The way I see to move forward is:

- make a new field in the table metadata (`maxValueSize`)
- In `OracleDdlTable.java`, instead of calculating `needsOverflow = metadata.getColumns().getMaxValueSize() > 2000;` from just the columns in the metadata, instead make it `needsOverflow = if (table.maxValueSize is set) { return table.maxValueSize > 2000 } else { return columns.maxValueSize > 2000 }`

Be aware this tangentially touches your existing customers on Postgres DbKVS and C*KVS because we're going to re-store the table metadata on those too to include this extra field. It should be harmless, though, and this should happen when we re-create() all schemas' tables on startup. It would /not/ be harmless if you already had Oracle DbKVS customers.
Also I'm not sure if we've tested the case (in OSS, this works in PG because of upgrade task framework) where you push in new metadata with additive change, revert back to old atlas version, and then startup again. Ideally it should re-store the "new" metadata with one less field, but if it tries to deserialize the existing metadata first it might freak out. Could possibly want to test this.

In PG, this needs doesnt-_need_-to-but-should have an upgrade task, to alter the main tables to remove or add some overflow columns and drop or create the extra overflow tables, all to re-assert the same state that you would have been in if your DB had been created from the current code.

**Comment by jboreiko** [comment]
Just in terms of priority, really this only amounts to a performance problem for fields that might have user specified maxes under 2000 characters.  It's worth determining how many of those cases exist before investing in this work, since we should be able to count the ones that exist currently.  

Inversely we could consider fixing this and then support people wishing to migrate to more efficient versions by specifying a max where they hadn't before.  That said if I had to guess I wouldn't imagine many OSS users could make use of this functionality.

**Comment by clockfort** [comment]
Yes, it's very likely just (older internal project) who has some maxValueSizes set that aren't doing anything in her schema's table metadata.
